Source/Events.Store.MongoDB/Processing/Streams/UnsupportedTypeForStreamProcessorStateDiscriminatorConvention.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Dolittle.Runtime.Events.Store.MongoDB.Processing.Streams
+{
+    /// <summary>
+    /// Exception that gets thrown when trying to use an unsupported type in  <see cref="StreamProcessorStateDiscriminatorConvention"/>.
+    /// </summary>
+    public class UnsupportedTypeForStreamProcessorStateDiscriminatorConvention : EventStoreConsistencyError
```

Comment:
This isn't really an event store consistency problem. This is something we have done wrong in the code. So just leave it with `: Exception`

Reply:
Fixd.